### PR TITLE
Battery: add a test loading an iframe page that is NOT about:blank

### LINKED ISSUE #3107: The battery-status/battery-promise.html test is incorrect
The "setting iframe's src makes its Navigator object vary thus getting another battery promise" creates an iframe with no src attribute, waits for onload, then sets src.  It then asserts that this should create a new Navigator and battery promise.

But per spec inserting the iframe will land in https://html.spec.whatwg.org/multipage/browsers.html#creating-a-new-browsing-context and create an about:blank document in the iframe, then go to https://html.spec.whatwg.org/multipage/embedded-content.html#process-the-iframe-attributes and thence to https://html.spec.whatwg.org/multipage/embedded-content.html#iframe-load-event-steps to fire the load event.

Now setting the src will navigate, but the navigation is same-origin and therefore the existing Window gets reused.  Hence there is no new Navigator in this case.

### PR DESCRIPTION
**Pull request by zqzhang**
Also split the tests checking a battery promise per navigator because
onmessage is used in both iframe and new window scenarios.

Resolve #3107.

### CODE REVIEW

**Comment by zqzhang** [comment]
@bzbarsky @Honry PTAL, thanks.

File: battery-status/battery-promise-iframe.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Battery Test: each Navigator object has a battery promise for BatteryManager</title>
```
Comment (Honry) [inline battery-status/battery-promise-iframe.html]: It's better to use different title for each promise test.

Comment (zqzhang) [inline battery-status/battery-promise-iframe.html]: I guess you mean test files' titles. The tests' names have been specified individually.

**Comment by Honry** [comment]
LGTM

**Comment by bzbarsky** [comment]
@zqzhang The battery-promise-iframe.html test is racy. There is nothing in the spec that requires the script to run, and the message listener to be added, before support-iframe-initial.html has loaded, done its postMessage, and had the message event fire.

You should be adding this listener before you start the load of support-iframe-initial.html.

**Pull request merged**